pvcompare/demand.py

```
@@ -105,28 +113,66 @@ def calculate_load_profiles(
     for column in energyConsumption:
         if column != "unit":
             if energyConsumption.at["energyVector", column] == "Heat":
-                calculate_heat_demand(
-                    country=country,
-                    lat=lat,
-                    lon=lon,
-                    storeys=storeys,
-                    year=year,
-                    weather=weather,
-                    static_inputs_directory=static_inputs_directory,
-                    user_inputs_pvcompare_directory=user_inputs_pvcompare_directory,
-                    user_inputs_mvs_directory=user_inputs_mvs_directory,
-                    column=column,
-                )
+
+                if add_heat_demand is not None:
+                    # check if file exists
+                    if os.path.isfile(add_heat_demand):
+                        # save the file name of the time series
+                        # mvs_inputs/elements/csv/energyProduction.csv
+                        check_inputs.add_file_name_to_energy_consumption_file(
+                            column=column,
+                            ts_filename=add_heat_demand,
+                            user_inputs_mvs_directory=user_inputs_mvs_directory,
+                        )
+                    else:
+                        logging.warning(
```

Careful: is the demand calculated here, if the file does not exist?

---

Yes. and a warning is given. You want to raise an error instead?